src/reducers/toyReducer.js

```diff
@@ -27,9 +27,9 @@ export default function toyReducer(state = initialState, action) {
       return { ...state, toys: deletetoys };
     case "DELETE_TOY":
       let DeleteToys = state.toys.filter(
-        (toy) => toy.id !== action.payload.toString()
+        (toy) => toy.id != action.payload.toString()
```

Comment:
Can you explain why you made this change?  (and perhaps there's a different change you can make to this line instead of the one you did).

Also, is there any reason why you have `DeleteToys` starting with a capital D here?

Reply:
A couple more things:

1.  `DeleteToys` isn't being changed, so should be a `const` not a `let`.
2. `DeleteToys` doesn't convey what's actual in the variable, it's actually all the toys except the one that was just deleted.  I think `toys` will be just fine, then you don't have to add a key:

```
  return { ...state, toys };
```

Reply:
Can you also take a look at the "DELETE_REVIEW" reducer?

```
case "DELETE_REVIEW":
      let deletetoys = state.toys.map((toy) => {
        if (toy.id === action.payload.id) {
          return action.payload;
        } else {
          return toy;
        }
      });
      return { ...state, toys: deletetoys };
```
Same things as above:

1. `deletetoys` isn't a good naming convention here, should be `deleteToys` (camel-case), but doesn't convey what actually is there, so really should just be `toys`.
2. It should be a `const` not a `let` since you're not changing it.

Reply:
> Can you explain why you made this change? (and perhaps there's a different change you can make to this line instead of the one you did).
> 
> Also, is there any reason why you have `DeleteToys` starting with a capital D here?

I made it to see if it would work too,  but I see now that it should be `(toy) => toy.id !== action.payload.toString()`.

It should definitely be camelcase.


> A couple more things:
> 
> 1. `DeleteToys` isn't being changed, so should be a `const` not a `let`.
> 2. `DeleteToys` doesn't convey what's actual in the variable, it's actually all the toys except the one that was just deleted.  I think `toys` will be just fine, then you don't have to add a key:
> 
> ```
>   return { ...state, toys };
> ```

 I actually went ahead and did this a different way because I could not get it to work like that. Is this acceptable code? It does not pass the tests, but it does work and it is cleaner, right?

```
    case "DELETE_REVIEW":
      state.toys.map((toy) => {
        if (toy.id === action.payload.id) {
          return action.payload;
        } else {
          return toy;
        }
      });
      return {...state,  toys: state.toys };
    case "DELETE_TOY":
       state.toys.filter(
        (toy) => { return toy.id !== action.payload.toString()}
      );
      return {...state, toys: state.toys };
    default:
      return state;
  }
}
```

Reply:
> Can you explain why you made this change? (and perhaps there's a different change you can make to this line instead of the one you did).
> 
> Also, is there any reason why you have DeleteToys starting with a capital D here?
> 
> I made it to see if it would work too, but I see now that it should be (toy) => toy.id !== action.payload.toString().
> 
> It should definitely be camelcase.

Do you know what `action.payload.toString()` is doing?

Reply:
> I actually went ahead and did this a different way because I could not get it to work like that. Is this acceptable code? It does not pass the tests, but it does work and it is cleaner, right?
> 
>     case 'DELETE_REVIEW':
>       state.toys.map((toy) => {
>         if (toy.id === action.payload.id) {
>           return action.payload;
>         } else {
>           return toy;
>         }
>       });
>       return { ...state, toys: state.toys };
>     case 'DELETE_TOY':
>       state.toys.filter((toy) => {
>         return toy.id !== action.payload.toString();
>       });
>       return { ...state, toys: state.toys };
>     default:
>       return state;

Are you sure this works?  Check again, because all you are doing here is returning the state.

Reply:
For example:

```
case 'DELETE_REVIEW':
      state.toys.map((toy) => {
        if (toy.id === action.payload.id) {
          return action.payload;
        } else {
          return toy;
        }
      });

     // state.toys.map is just going to iterate through the state.toys, it doesn't change the state.toys.

      return { ...state, toys: state.toys };
     // ...state destructures the whole state object, which includes toys, so there is no reason to set toys to state.toys, this doesn't do anything.
```

Reply:
https://jsfiddle.net/krjamcw0/

Reply:
Hey Shawn, what do you think about this? Also yes, I don't know why I was set on trying to convert it  back to a string hahaha. 

```
export const initialState = {
  toys: [],
};

export default function toyReducer(state = initialState, action) {
  let { toys } = state;
  switch (action.type) {
    case 'FETCH_TOYS':
      return { toys: action.payload };
    case 'CREATE_TOY':
      return { ...state, toys: [...toys, action.payload] };
    case 'CREATE_REVIEW':
      return {
        ...state,
        toys: toys.map((toy) =>
          toy.id == action.payload.id ? action.payload : toys
        ),
      };
    case 'DELETE_REVIEW':
      toys = toys.map((toy) => {
        if (toy.id === action.payload.id) {
          return action.payload;
        } else {
          return toy;
        }
      });
      return { ...state, toys };
    case 'DELETE_TOY':
      toys = toys.filter((toy) => {
        return toy.id !== action.payload;
      });
      return { ...state, toys };
    default:
      return state;
  }
}
```

Reply:
For readability, I'd lose the `let { toys } = state` and just go with:
```
const toys = state.toys.map(...);
return  { ...state, toys };
```

and

```
const toys = state.toys.filter(...);
return { ...state, toys };
```

for CREATE_REVIEW, it's getting a bit busy, and is probably easier readable by doing the same as DELETE_REVIEW and DELETE_TOY:
```
const toys = state.toys.map(...);
return { ...state, toys };
```

The CREATE_TOY is pretty concise, so think leaving it alone is fine.

Reply:
Also remember that you can omit the `return` statement if you don't use `{}`.  Also you don't need to use parenthesis if your function only has a single argument:


```
...
case 'DELETE_TOY':
      const toys = state.toys.filter(toy => toy.id !== action.payload);
      return { ...state, toys };
...

Reply:
You can also use ternary operator as well, as long as it results in readable code:
 
```
case 'DELETE_REVIEW':
  const toys = state.toys.map(toy => toy.id === action.payload.id ? action.payload : toy);
  return { ...state, toys };
```

Reply:
Hey Shawn, what do you think about this? I cannot define toys with const more than once, is this okay?

```
export const initialState = {
  toys: [],
};

export default function toyReducer(state = initialState, action) {
  let toys;
  switch (action.type) {
    case 'FETCH_TOYS':
      return { toys: action.payload };
    case 'CREATE_TOY':
      return { ...state, toys: [...state.toys, action.payload] };
    case 'CREATE_REVIEW':
      toys = state.toys.map(toy =>
        toy.id == action.payload.id ? action.payload : toys
      );
      return { ...state, toys };
    case 'DELETE_REVIEW':
      toys = state.toys.map(toy => toy.id === action.payload.id ? action.payload : toy);
      return { ...state, toys };
    case 'DELETE_TOY':
      toys = state.toys.filter(toy => toy.id !== action.payload);
      return { ...state, toys };
    default:
      return state;
  }
}

```

Reply:
I went ahead and I pushed a new commit to this PR, I also have the `useEffect ` for the toycontainer because I had an endless loop happening in my code.